hedera-node/src/main/java/com/hedera/services/grpc/marshalling/ImpliedTransfersMarshal.java

```diff
@@ -66,17 +74,117 @@ public ImpliedTransfers unmarshalFromGrpc(CryptoTransferTransactionBody op) {
 		}
 
 		final List<BalanceChange> changes = new ArrayList<>();
+		final List<Pair<EntityId, List<CustomFee>>> entityCustomFees = new ArrayList<>();
+		List<CustomFeesBalanceChange> customFeeBalanceChangesForRecord = new ArrayList<>();
 		for (var aa : op.getTransfers().getAccountAmountsList()) {
 			changes.add(hbarAdjust(aa));
 		}
+		var payerId = EntityId.fromGrpcAccountId(payer);
 		for (var scopedTransfers : op.getTokenTransfersList()) {
 			final var grpcTokenId = scopedTransfers.getToken();
-			final var scopingToken = Id.fromGrpcToken(grpcTokenId);
+			final var scopingToken = EntityId.fromGrpcTokenId(grpcTokenId);
+			var amount = 0L;
 			for (var aa : scopedTransfers.getTransfersList()) {
 				changes.add(tokenAdjust(scopingToken, grpcTokenId, aa));
+				if (aa.getAmount() > 0) {
```

Comment:
Should we also consider the amount transfer in to the payer account? For a hypothetical scenario, the payer initially has 100 A token. In a transfer transaction that will transfer 200 A token from account X to payer, and transfer 250 A token to account Y. From the payer's view point, the transaction should go through. But the current logic may report insufficient balance error.

Reply:
Great point! This is addressed by consolidating all `(account, token/ℏ)` balance changes via the `existingBalanceChanges` map.